Makefile

```diff
@@ -0,0 +1,30 @@
+.PHONY: all clean lint type test test-cov
```

Comment:
Is there a reason to migrate over to a `Makefile`? The bash scripts were working well, and a make file is definitely the wrong style to use with this. 

Can we restore the bash scripts.

Reply:
f183f13